clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -446,4 +463,30 @@ object GarbageCollector {
 
     bulkRemove(df, MaxBulkSize, bucket, region, awsRetries, snPrefix, hcValues).toDF("addresses")
   }
+
+  private def writeParquetReport(
+      dstRoot: String,
+      df: DataFrame,
+      time: String,
+      suffix: String = ""
+  ) = {
+    val dstPath = dstRoot + s"/dt=${time}/${suffix}"
+    df.write.parquet(dstPath)
+  }
+
+  private def writeJsonSummary(
+      dstRoot: String,
+      numDeletedObjects: Long,
+      gcRules: String,
+      hcValues: Broadcast[ConfMap],
+      time: String
+  ) = {
+    val dstPath = new Path(dstRoot + s"/dt=${time}/summary.json")
+    val dstFS = dstPath.getFileSystem(configurationFromValues(hcValues))
+    val jsonSummary = JObject("gc_rules" -> gcRules, "num_deleted_objects" -> numDeletedObjects)
```

Comment:
This is a bit strange: we just got `gcRules`, a value that is explicitly JSONable.  We transformed it into a string and then include this stringified value inside our real JSON.  The problem is that JSON-in-JSON is pretty awful to read (every quote is backslashed, newlines added for formatting before less readable, etc.).  Can we instead just stuff gcRules in some JSONified form here?